spec/acceptance/securitygroup_spec.rb

```diff
@@ -378,4 +389,91 @@ def expect_rule_matches(ingress_rule, ip_permission)
 
   end
 
+  describe 'should create a new security group' do
+
+    before(:all) do
+      @name = "#{PuppetManifest.env_id}-#{SecureRandom.uuid}"
+      @config = {
+        :name => @name,
+        :ensure => 'present',
+        :description => 'aws acceptance securitygroup',
+        :region => @default_region,
+        :ingress => [
+          {
+            :protocol  => 'tcp',
+            :from_port => 22,
+            :to_port => 1000,
+            :cidr      => '0.0.0.0/0'
+          }
+        ],
+      }
+
+      PuppetManifest.new(@template, @config).apply
+      @group = get_group(@config[:name])
+    end
+
+    after(:all) do
+      new_config = @config.update({:ensure => 'absent'})
+      PuppetManifest.new(@template, new_config).apply
+
+      expect { get_group(@config[:name]) }.to raise_error(::Aws::EC2::Errors::InvalidGroupNotFound)
+    end
+
+    it "with the specified ingress rules" do
+      @config[:ingress].all? { |rule| has_ingress_rule(rule, @group.ip_permissions)}
+    end
+
+    rules_to_test = [
+      [{
+        :protocol  => 'udp',
+        :from_port => 80,
+        :to_port   => 100,
+        :cidr      => '0.0.0.0/0'
+      }],
+      [{
+        :port => 22,
+        :cidr => '0.0.0.0/0'
+      }],
+      [{
+        :protocol => 'tcp',
+        :port     => 22,
+        :cidr     => '0.0.0.0/0'
+      },{
+        :protocol  => 'udp',
+        :from_port => 80,
+        :to_port   => 100,
+        :cidr      => '0.0.0.0/0'
+      }],
+      [{
+        :protocol => 'tcp',
+        :security_group => '<<name>>',
+      }],
+      [{
+        :protocol => 'udp',
+        :from_port => 100,
+        :to_port   => 120,
+        :security_group => '<<name>>',
+      }]
+    ]
+
+    rules_to_test.each_with_index do |rules, i|
+      it "should be able to modify the ingress rules protocol and ports: #{i+1}" do
+        new_rules = []
+        rules.each do |rule|
+          rule[:security_group] = @name if rule.keys.include?(:security_group)
+          new_rules << rule
+        end
+        new_config = @config.dup.update({:ingress => new_rules})
+        success = PuppetManifest.new(@template, new_config).apply[:exit_status].success?
+        expect(success).to eq(false)
```

Comment:
Nope, because we're changing results and receiving detailed exist codes. We should probably assert on the exist_status being 2 however.